File: java/src/main/behavioral/chainofresponsibility/ConcreteHandlerB.java
package behavioral.chainofresponsibility;

// ConcreteHandlerB: Handles certain requests and passes the rest
public class ConcreteHandlerB extends Handler {
    @Override
    public void handleRequest(String request) {
        if (request.equalsIgnoreCase("B")) {
            System.out.println("ConcreteHandlerB handled request: " + request);
        } else if (nextHandler != null) {
            nextHandler.handleRequest(request);
        } else {
            System.out.println("No handler could process request: " + request);
        }
    }
}

File: python/src/behavioral/chain_of_responsibility/concrete_handler_b.py
from .abstract_handler import AbstractHandler


class ConcreteHandlerB(AbstractHandler):
    """
    Handles requests starting with 'B'.
    """

    def handle(self, request: str) -> str:
        if request.startswith("B"):
            return f"Handler B processed: {request}"
        return super().handle(request)


# Example Usage:
#
# h1 = ConcreteHandlerA()
# h2 = ConcreteHandlerB()
# h1.set_next(h2)
#
# print(h1.handle("Apple"))   # Handler A processed: Apple
# print(h1.handle("Banana"))  # Handler B processed: Banana
# print(h1.handle("Cherry"))  # None
#
#
# Common Mistakes to Avoid:
#
# 1. Forgetting to forward to the next handler.
#    - Fix: When handler cannot process, call super().handle() or
#      next_handler.handle().
#
# 2. Hardcoding all logic into one handler.
#    - Fix: Each handler should handle only one specific type of request.
#
# 3. Returning errors instead of None for unhandled requests.
#    - Fix: Return None to indicate the chain did not process the request.
